custom/abdm/milestone_one/utils/decorators.py

```diff
@@ -0,0 +1,37 @@
+import logging
+
+from functools import wraps
+from typing import List
+
+from custom.abdm.milestone_one.utils.response_util import generate_invalid_req_response
+
+logger = logging.getLogger(__name__)
+
+
+def required_request_params(params):
+    """
+    Checks if the parameters provided in the decorator(a list of strings) are present in the DRF request.
+    If not, raises 400 Bad Request error.
+    """
+
+    def decorate(fn):
+        if not (params and isinstance(params, List)):
```

Comment:
To mandate passing mandatory params in a list, check is added here.

Reply:
but this would fail later anyway if params are missing?

Reply:
Mandatory params are defined by us while exposing an endpoint. So, these params are not what come in request but are those against which request is verified if those param-value combo is present in the req. or not.

Reply:
Not sure I understand. My original comment was that we were checking if any params were passed at all, which seems unnecessary. Responding back with "No params passed" or "Missing params: X, Y, Z", are the same things from HQ perspective.

Reply:
This is a check that params added in the decorator of view is a list(E.g. https://github.com/dimagi/commcare-hq/pull/32332/files/9e1cf032f3772e2f31bbb39681f46819206e3e93#diff-d8c9bf091e1c3c337093c95f36be1b44da3799bec9b511e06db61349142ca886R13). This does not check the params passed in the request. Are we on the same understanding here?

Reply:
> This is a check that params added in the decorator of view is a list(

Right! I didn't get that. In that case, can the error message be updated? This error would come when HQ envrionment is loaded during startup or it will only happen when the view is being called

Reply:
It is used during view call.
https://github.com/dimagi/commcare-hq/pull/32332/commits/3d2192483a02e0bee4b0d4402f8a59f0fce8cbcc